server-backend/src/NbxplorerService.ts

```diff
@@ -191,57 +304,75 @@ export class NbxplorerService implements OnApplicationBootstrap, OnApplicationSh
-        const response = await (await fetch(`${this.config.baseUrl}/derivations/${xpub}/balance`)).json();
+    getUrl(cryptoCode: string = 'btc'): string {
+        return `${this.config.baseUrl}/${cryptoCode}`;
+    }
+
+    async getBalance(xpub: string, cryptoCode: string = 'btc'): Promise<NBXplorerBalance> {
+        const response = await (await fetch(`${this.getUrl(cryptoCode)}/derivations/${xpub}/balance`)).json();
         return nbxplorerBalanceSchema.parse(response);
     }
 
-    async track(xpub: string): Promise<void> {
-        const response = await fetch(`${this.config.baseUrl}/derivations/${xpub}`, { method: 'POST' });
+    async track(xpub: string, cryptoCode: string = 'btc'): Promise<void> {
+        const response = await fetch(`${this.getUrl(cryptoCode)}/derivations/${xpub}`, { method: 'POST' });
         if (response.status >= 300) {
             throw new Error('nbxplorer threw an error when tracking xpub');
         }
     }
 
-    async trackAddress(address: string): Promise<void> {
-        const response = await fetch(`${this.config.baseUrl}/addresses/${address}`, { method: 'POST' });
+    async trackAddress(address: string, cryptoCode: string = 'btc'): Promise<void> {
+        const response = await fetch(`${this.getUrl(cryptoCode)}/addresses/${address}`, { method: 'POST' });
         if (response.status >= 300) {
-            throw new Error('nbxplorer threw an error when tracking xpub');
+            throw new Error(`nbxplorer threw an error when tracking address: ${address}`);
         }
     }
 
-    async getUnusedAddress(xpub: string, opts?: {
+    async getUnusedAddress(xpub: string, cryptoCode: string = 'btc', opts?: {
         change?: boolean,
-        reserve?: boolean
+        reserve?: boolean,
     }): Promise<NBXplorerAddress> {
         const change = opts?.change ?? false;
         const reserve = opts?.reserve ?? false;
         const params = new URLSearchParams({
             feature: change ? 'Change' : 'Deposit',
             reserve: reserve.toString(),
         });
-        const response = await (await fetch(`${this.config.baseUrl}/derivations/${xpub}/addresses/unused?${params}`)).json();
+        const response = await (await fetch(`${this.getUrl(cryptoCode)}/derivations/${xpub}/addresses/unused?${params}`)).json();
         return nbxplorerAddressSchema.parse(response);
     }
-    async getUTXOs(xpub: string): Promise<NBXplorerUtxosResponse> {
-        const response = await (await fetch(`${this.config.baseUrl}/derivations/${xpub}/utxos`)).json();
+
+    async generateHotWallet(cryptoCode: string = 'btc'): Promise<nbxplorerHotWallet> {
+        const response = await (await fetch(`${this.getUrl(cryptoCode)}/derivations`, {
+            method: 'POST',
+        })).json();
+        return nbxplorerHotWalletSchema.parse(response);
+    }
+
+    async getUTXOs(xpub: string, cryptoCode: string = 'btc'): Promise<NBXplorerUtxosResponse | void> {
```

Comment:
we are never returning void
```suggestion
    async getUTXOs(xpub: string, cryptoCode: string = 'btc'): Promise<NBXplorerUtxosResponse> {
```

Reply:
Removed since its not needed